### PR TITLE
Pipeline, Yulia Vodopyanova - 21FPL1 

### PR DESCRIPTION
**Pull request by YuliaVodopyanova**

### CODE REVIEW

File: lab_5_scrapper/target_score.txt [hunk 1]
```diff
@@ -1 +1 @@
-6
\ No newline at end of file
+10
```
Comment (demid5111) [inline lab_5_scrapper/target_score.txt]: are you planning to increase the mark? now you have got 8 according to the spreadsheet

**Review by demid5111 — CHANGES_REQUESTED** [review]
nothing to review regarding pipeline. question about higher mark for scrapper - do you plan to attempt for 10?

File: lab_6_pipeline/pipeline.py [hunk 2]
```diff
@@ -18,21 +33,56 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+
+        meta_files = [meta for meta in self.path_to_raw_txt_data.glob(r'*_meta.json')]
+        raw_files = [raw for raw in self.path_to_raw_txt_data.glob(r'*_raw.txt')]
+
+        if len(meta_files) != len(raw_files):
+            raise InconsistentDatasetError
+
+        raw_ids = sorted(get_article_id_from_filepath(raw) for raw in raw_files)
+        try:
+            ids_order = list(range(1, raw_ids[-1] + 1))
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:61]: you need to check that numbering is 1...N without slips

File: lab_6_pipeline/pipeline.py [hunk 3]
```diff
@@ -18,21 +33,56 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+            if ids_order != raw_ids:
+                raise InconsistentDatasetError
+        except IndexError:
+            pass
+
+        for files in meta_files, raw_files:
+            if not all(file.stat().st_size for file in files):
+                raise InconsistentDatasetError
+
+        if not self.path_to_raw_txt_data.iterdir() or \
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: what is the reason for the second check with `any`

File: lab_6_pipeline/pipeline.py [hunk 4]
```diff
@@ -75,6 +126,8 @@ def get_cleaned(self) -> str:
         """
         Returns lowercase original form of a token
         """
+        return self._text.translate(str.maketrans(
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py]: rework for regex for simplicity

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+beautifulsoup4==4.12.0
+lxml==4.9.2
+requests==2.28.2
+pymystem3==0.2.0
```
Comment (demid5111) [inline requirements.txt]: where are you using it?

**Review by demid5111 — CHANGES_REQUESTED** [review]
continue! time is approaching

**Review by demid5111 — CHANGES_REQUESTED** [review]
nothing new to review. hurry up, your PR is at risk of missing deadline

File: requirements.txt (hunk 5 above)
Comment (YuliaVodopyanova) [reply requirements.txt]: It was for the future

File: lab_5_scrapper/target_score.txt (hunk 1 above)
Comment (YuliaVodopyanova) [reply lab_5_scrapper/target_score.txt]: nono i just forgot to change it

File: lab_6_pipeline/pipeline.py (hunk 4 above)
Comment (YuliaVodopyanova) [reply lab_6_pipeline/pipeline.py]: done

File: lab_6_pipeline/pipeline.py [hunk 6]
```diff
@@ -18,21 +35,55 @@ def __init__(self, path_to_raw_txt_data: Path):
...
+        raw_ids = sorted(get_article_id_from_filepath(raw) for raw in raw_files)
+        try:
+            ids_order = list(range(1, raw_ids[-1] + 1))
+            if ids_order != raw_ids:
+                raise InconsistentDatasetError
+        except IndexError:
+            pass
+
+        for files in meta_files, raw_files:
+            if not all(file.stat().st_size for file in files):
```
Comment (demid5111) [inline lab_6_pipeline/pipeline.py:68]: what is all and any? how do they work?

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**